package/network/services/lldpd/files/lldpd.init

```diff
@@ -81,9 +81,25 @@ get_config_cid_ifaces() {
 
 	local _iface _ifnames=""
 	for _iface in $_ifaces; do
-		local _ifname=""
-		if network_get_device _ifname "$_iface" || [ -e "/sys/class/net/$_iface" ]; then
-			append _ifnames "${_ifname:-$_iface}" ","
+		local _l3dev=""
+		# save any "!" or "!!" prefix from the interface name
+		_suffix=${_iface##*"!"}
+		_prefix=${_iface%%"$_suffix"}
+
+		if network_get_device _l3dev "$_suffix" || [ -e "/sys/class/net/$_suffix" ]; then
+			if [ -n "$_prefix" ]; then
+				# prepend the stripped "!" or "!!" prefix here if we had one
+				append _ifnames "$_prefix${_l3dev:-$_suffix}" ","
+			else
+				append _ifnames "${_l3dev:-$_suffix}" ","
+			fi
+		else
+			case $_iface in 
+				*"*"*)
```

Comment:
Is the third `*` necessary?  I think you only want a `*` suffix for globbing, don't you?

Reply:
I'm not sure - consider the patterns: `!*:*`, `!eth*` or even `eth*1`.

Reply:
okay. as long as `lldpd` itself can handle those patterns.

Reply:
OK. Yeah, that's the raison d'être for this PR. lldpd handles those.